### DISCUSSION TITLE
Why i cant use composables inside composables callback?

### DISCUSSION THREAD

**Discussion by Venegrad** [nuxt/nuxt#26173] [General]
https://stackoverflow.com/questions/78134978/how-to-create-usefetch-wrapper-on-nuxtjs3?noredirect=1#comment137748916_78134978

```
const data = useFetch(link, {
    ...params,
    onResponse: ({ request, response, options }) => {
       useLogout()
       navigateTo('/')
    },
    onResponseError: ({response}) => {
      useLogout()
      navigateTo('/')
    }
  })
```

In this case 
``` 
useLogout()
      navigateTo('/')
```
Navigate or any composables not working from ssr, but working on client, can i ask why? and how to solve propblem, when i need to wrapping error to logout user, but i cant use useFetch() ? cuz i cant handle errors ?

**Comment by huang-julien** [comment]
Hi that's because the context is unset after calling your composable. You'd need to use `nuxtApp.runWithContext` to re-set the nuxtApp instance when calling composables in the callback.

It's working client-side because the nuxt instance is considered as a Singleton. While server-side we can have multiple request in concurrence meaning multiple nuxtApp running at the same time
